### ISSUE TITLE
Cannot declare class CreateJobsTable, because the name is already in use

### ISSUE THREAD

**Issue by steros** [laravel/framework#23194]
- Laravel Version: 5.5
- PHP Version: 7.1
- Database Driver & Version: sqlite (:memory:)

### Description:
If I use the trait "Illuminate\Foundation\Testing\RefreshDatabase" as proposed on the docs I get the error:
Fatal error: Cannot declare class CreateJobsTable, because the name is already in use in {root}/database/migrations/{timestamp}_create_jobs_table.php on line 36

### Steps To Reproduce:
1. php artisan queue:table
2. php artisan migrate
3. setup a test like: https://laravel.com/docs/5.5/database-testing#resetting-the-database-after-each-test
4. create a factory
4. create a database table seeder using that factory
5. put the seeder into the DatabaseSeeder.php's run method
4. run the test

**Comment by Marwelln** [comment]
I have the same problem but trying to use observers.

1. Creating a new observer in `App\Observers` as `FooObserver`.
2. Register observer in `boot` method of `AppServiceProvider`:  `MyModel::observe(FooOserver::class);`
3. `"Cannot declare class App\Oberserver\FooObserver, because the name is already in use"`

**Comment by dwightwatson** [comment]
The answer is in the error message - you are declaring a class that already exists in the given namespace. Search your project to find where the class name is used. It's likely you've reused a generator and simply generated duplicate classes.

**Comment by steros** [comment]
@dwightwatson Of course the answer is in the error message. Except I don't declare a class of that name. The only thing that created a class of that name was `php artisan queue:table`. It's the only place that class exists. So the real answer for the error is hidden in the trait, because using that causes the error to appear...

**Comment by ngkarkeong** [comment]
I have the similar issue, my situation is i try to move my listener "thanksmail", to "mail/thanksmail". after i tried to clear cache, clear config and compose autodump, i still send me the same erroe, class thanksmail is used.

**Comment by levidurfee** [comment]
I know this is closed, but this is the only resource I can find where people are having the same issue.

**Comment by xyznaveen** [comment]
Possible Solution:  Check the database migration folder for duplicate **create_jobs_table** and simply remove it.

I accidentally ran **php artisan queue:table** twice which lead to this error. Also realized this only after I read @levidurfee's answer.

Hope this helps someone.

**Comment by bezner** [comment]
Check the namespace, if your PSR-4 namespace won't match the class file location you will likely get that error.  Run composer dump-autoload afterwards to be on a save side.

**Comment by raselkarim7** [comment]
> Totally Agree with @xyznaveen, I also faced the same problem and implemented same solution. :)

**Comment by guigranzeol** [comment]
I deleted the vendor path, and after I ran composer install.

This way worked in my case

**Comment by kapitoxapux** [comment]
Maybe it will help someone... in my case there were two migrations "create_failed jobs_table", don't know how that happen but that was the reason

**Comment by ffmit** [comment]
Same for me, it was a duplicate of the create_sessions_table in the migrations folder of Laravel.
Caused by miself when I manually migrated a project from L7 to L8 and there I accidentally moved the duplicate one from L7 to L8, where it was already existing.

**Comment by Volkan61** [comment]
for the people who have the same issue. you can search the table name in the scope in intellij than migrations included in libraries are searched as well. in my case CreateRolesTable was also included in the dependency jeremykenedy/laravel-roles and somehow these migrations were triggered as well when I called php artisan migrate. After I removed the depenency, the problem was solved! Or you should only use the dependency, and remove the duplicate migrations! Cheers!

**Comment by Volkan61** [comment]
and don't forget to recreate bootstrap/cache, because it contains the code which triggers the migration jeremykenedy/laravel-roles. But don't ask me why this was not trggered before lol

**Comment by haejinjo** [comment]
What fixed it for me was returning an anonymous class from the migration file.

So I replace this:  
```
class CreateEventsTable extends Migration {

    public function up() {
       Schema::create('examples', function (Blueprint $table) {
           ....
       });
...
};
```

with this:  

```
return new class extends Migration {

    public function up() {
       Schema::create('examples', function (Blueprint $table) {
           ....
       });
...
};
```
